src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -442,7 +442,7 @@ await foreach (var response in requests)
         private CSharpVirtualDocumentSnapshot? GetCSharpDocumentSnapshsot(Uri uri)
         {
             var normalizedString = uri.GetAbsoluteOrUNCPath();
-            var normalizedUri = new Uri(WebUtility.UrlDecode(normalizedString));
+            var normalizedUri = new Uri(normalizedString);
```

Comment:
`uri.GetAbsoluteOrUNCPath()` already decodes once. By calling `new Uri(WebUtility.UrlDecode(normalizedString))` we incorrectly decode an additional time.